### PR TITLE
Improved: Added separate views for fetching Fulfilled Order Item and Shipment Details instead of single view (#34)

### LINKED ISSUE #34: Refactor OrderItemAndShipmentSyncQueue View to separate Shipment related entities
1. Refactor OrderItemAndShipmentSyncQueue View to separate Shipment related entities into separate view.
2. This view is bulky with lot of entities being included to fetch the Order Item and Shipment related information.
3. With bulk data set around ~ 3 Lac order item records or more, the view will take a lot of time to fetch the records.

### PR DESCRIPTION
**Pull request by yaminibhagat**
Closes #34 

1. Created the OrderItemAndShipmentSyncQueueNew for fetching the Order Item Level information.
2. Created the ShipmentItemView for fetching the Shipment Item Level information.

### CODE REVIEW

File: entity/OmsOrderViewEntities.xml [hunk 1]
```diff
@@ -412,6 +412,207 @@ under the License.
...
+    <view-entity entity-name="OrderItemAndShipmentSyncQueueNew" package="co.hotwax.warehouse" group="ofbiz_transactional">
+        <description>View to get details for the Order Item.</description>
+        <member-entity entity-alias="OH" entity-name="org.apache.ofbiz.order.order.OrderHeader"/>
+        <member-entity entity-alias="OI" entity-name="org.apache.ofbiz.order.order.OrderItem" join-from-alias="OH">
+            <key-map field-name="orderId"/>
+        </member-entity>
+        <member-entity entity-alias="SCENM" entity-name="org.apache.ofbiz.common.enum.Enumeration" join-from-alias="OH" join-optional="true">
+            <key-map field-name="salesChannelEnumId" related="enumId"/>
+        </member-entity>
+        <member-entity entity-alias="UOM" entity-name="org.apache.ofbiz.common.uom.Uom" join-from-alias="OH">
```
Comment (gurveenbagga) [inline entity/OmsOrderViewEntities.xml:426]: Can we fetch Uom details using cache in the service where the view is used ?

File: entity/OmsOrderViewEntities.xml [hunk 2]
```diff
@@ -412,6 +412,207 @@ under the License.
         </entity-condition>
     </view-entity>
 
+
+    <!--  View entity to fetch the Order Items for FulfilledOrder Reporting, using member entities of OFBiz  -->
+    <view-entity entity-name="OrderItemAndShipmentSyncQueueNew" package="co.hotwax.warehouse" group="ofbiz_transactional">
```
Comment (gurveenbagga) [inline entity/OmsOrderViewEntities.xml:417]: Please update view name

Comment (yaminibhagat) [reply entity/OmsOrderViewEntities.xml:417]: Sure ma'am,

File: entity/OmsOrderViewEntities.xml [hunk 3]
```diff
@@ -412,6 +412,216 @@ under the License.
...
+    <view-entity entity-name="ShipmentItemView" package="co.hotwax.warehouse" group="ofbiz_transactional">
+        <description>View to get Shipment details.</description>
+        <member-entity entity-alias="OI" entity-name="org.apache.ofbiz.order.order.OrderItem"/>
+        <member-entity entity-alias="OH" entity-name="org.apache.ofbiz.order.order.OrderHeader" join-from-alias="OI">
+            <key-map field-name="orderId"/>
+        </member-entity>
+        <member-entity entity-alias="SCENM" entity-name="org.apache.ofbiz.common.enum.Enumeration" join-from-alias="OH" join-optional="true">
+            <key-map field-name="salesChannelEnumId" related="enumId"/>
+        </member-entity>
+        <member-entity entity-alias="UOM" entity-name="org.apache.ofbiz.common.uom.Uom" join-from-alias="OH">
```
Comment (gurveenbagga) [inline entity/OmsOrderViewEntities.xml:504]: I think we should use join-optional here also

File: entity/OmsOrderViewEntities.xml [hunk 4]
```diff
@@ -412,6 +412,216 @@ under the License.
...
+                <econditions combine="or">
+                    <econdition entity-alias="OISGA" field-name="quantity" operator="greater" to-entity-alias="OISGA" to-field-name="cancelQuantity"/>
+                    <econdition entity-alias="OISGA" field-name="cancelQuantity" operator="equals" value=""/>
+                </econditions>
+            </econditions>
+            <order-by field-name="entryDate"/>
+        </entity-condition>
+    </view-entity>
+
+    <view-entity entity-name="ShipmentItemView" package="co.hotwax.warehouse" group="ofbiz_transactional">
```
Comment (gurveenbagga) [inline entity/OmsOrderViewEntities.xml:495]: This is not only Shipment details, can we rename this to "OrderItemAndShipment"

**Pull request merged**